### PR TITLE
GDPR-147: Add offender restriction check

### PR DESCRIPTION
**Pull request by hyderkhanmoj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/nomis/datacompliance/events/listeners/DataComplianceEventListener.java [hunk 1]
```diff
@@ -129,13 +136,29 @@ private void handleFreeTextMoratoriumCheck(final Message<String> message) {
         freeTextSearchService.checkForMatchingContent(event.getOffenderIdDisplay(), event.getRetentionCheckId(), event.getRegex());
     }
 
+    private void handleOffenderRestrictionCheck(Message<String> message) {
```
Comment (brightonsbox) [inline src/main/java/uk/gov/justice/hmpps/nomis/datacompliance/events/listeners/DataComplianceEventListener.java:139]: Again here just minor comment on final imput params

File: src/main/java/uk/gov/justice/hmpps/nomis/datacompliance/events/publishers/DataComplianceAwsEventPusher.java [hunk 2]
```diff
@@ -79,10 +80,17 @@ public void sendDuplicateDataResult(final DataDuplicateResult event) {
...
+        log.trace("Sending offender restriction result for offender: {}", event.getOffenderIdDisplay());
+
+        sqsClient.sendMessage(generateRequest("DATA_COMPLIANCE_OFFENDER-RESTRICTION-RESULT", event));
+    }
+
     @Override
     public void send(final FreeTextSearchResult event) {
 
-        log.trace("Sending duplicate data result for offender: {}", event.getOffenderIdDisplay());
+        log.trace("Sending free text search result for offender: {}", event.getOffenderIdDisplay());
```
Comment (brightonsbox) [inline src/main/java/uk/gov/justice/hmpps/nomis/datacompliance/events/publishers/DataComplianceAwsEventPusher.java:93]: 👍

**Review by brightonsbox — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/hmpps/nomis/datacompliance/events/listeners/DataComplianceEventListener.java (hunk 1 above)
Comment (hyderkhanmoj) [reply src/main/java/uk/gov/justice/hmpps/nomis/datacompliance/events/listeners/DataComplianceEventListener.java]: updated

**Review by brightonsbox — APPROVED** [review]
LGTM

**Pull request merged**